core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpWriterJob.java

```diff
@@ -87,34 +87,34 @@ public boolean run(int workerId) {
         assert this.workerId == workerId;
         boolean busy = drainQueue();
         // while ILP is hammering the database via multiple connections the writer
-        // is likely to be very busy so commitIfTimeoutElapsed() will run infrequently
+        // is likely to be very busy so checkIfTablesNeedCommit() will run infrequently
         // commit should run regardless the busy flag but has to finish quickly
         // idea is to store the tables in a heap data structure being the tables most
         // desperately need a commit on the top
-        if (!busy && !commitIfTimeoutElapsed()) {
+        if (!busy) {
+            checkIfTablesNeedCommit();
             tickWriters();
         }
         return busy;
     }
 
-    private boolean commitIfTimeoutElapsed() {
+    private void checkIfTablesNeedCommit() {
         final long millis = millisecondClock.getTicks();
-        if (millis > commitDeadLine) {
+        if (millis > nextCommitTime) {
+            long minTableNextCommitTime = Long.MAX_VALUE;
             for (int n = 0, sz = assignedTables.size(); n < sz; n++) {
-                if (assignedTables.getQuick(n).commitIfTimeoutElapsed(millis)) {
-                    // this return is to avoid holding up ingestion
-                    // after every successful commit we go back to drain the queue
-                    // if the queue is still empty we will come back and other tables will have their chance for a commit too
-                    // obviously tables at the end of the list (assignedTables) are disadvantaged but commits are
-                    // still issued automatically after we reached the max number of uncommitted rows and eventually
-                    // all tables in the list will get their commit after ingestion stops or slows down
-                    // the heap based solution mentioned above will eliminate this problem completely
-                    return true;
+                // the heap based solution mentioned above will eliminate the minimum search
+                // we could just process the min element of the heap until we hit the first commit
+                // time greater than millis and that will be our nextCommitTime
+                long tableNextCommitTime = assignedTables.getQuick(n).checkIfTableNeedsCommit(millis);
```

Comment:
checkIfTableNeedsCommit() is a confusing name, it does do commit whereas name doesn't lead you to believe it does. Can we rename ?

Reply:
done